x/bank/keeper/keeper.go

```diff
@@ -151,17 +151,8 @@ func (k BaseKeeper) DelegateCoins(ctx context.Context, delegatorAddr, moduleAccA
 	if err := k.trackDelegation(ctx, delegatorAddr, balances, amt); err != nil {
 		return errorsmod.Wrap(err, "failed to track delegation")
 	}
-	// emit coin spent event
-	delAddrStr, err := k.ak.AddressCodec().BytesToString(delegatorAddr)
-	if err != nil {
-		return err
-	}
-	sdkCtx := sdk.UnwrapSDKContext(ctx)
-	sdkCtx.EventManager().EmitEvent(
-		types.NewCoinSpentEvent(delAddrStr, amt),
-	)
```

Comment:
I think all token movements should emit an event. So, unless there is a duplicated event in this flow, we should keep the event.

Reply:
One benefit of having these events in here is that when reviewing historical block_results data it is trivial to track token movement between accounts.  While the state streaming solution is certainly an excellent approach it isn't useful for historical data. Further when reprocessing older chains the state streaming isn't an option.  Reprocessing data in block results is still possible though.

Reply:
We have done state streaming for historical information. We haven't open sourced the tool yet but will in the near future. Still open in draft to get feedback. Thank you for commenting.

Reply:
Extremely interested in what you have done for historical state streaming... that would be a gold standard solution (depending on how it ties in with older binaries).

Reply:
It works directly on iavl, the only thing which is annoying is if there are key or proto changes between binaries.